pkg/katautils/config.go

```diff
@@ -705,6 +705,14 @@ func updateConfig(configPath string, tomlConf tomlConfig, config *oci.RuntimeCon
 		return err
 	}
 
+	// vsock conflicts with factory, when both of them are enabled, kata will try
+	// to create a new vm template which is useless, thus it's better to disable
+	// template to save a lot of time.
+	if config.HypervisorConfig.UseVSock && config.FactoryConfig.Template {
+		kataUtilsLogger.Warn("vsock conflicts with factory, configure to not use it")
+		config.FactoryConfig.Template = false
```

Comment:
Should we just post an error and bail out here? Just so that we don't make users think template+vsock is working.